### DISCUSSION TITLE
Best way to align verticaly `text input` + `switch`

### DISCUSSION THREAD

**Discussion by cavasinf** [tabler/tabler#1390] [Q&A]
Hey,

I've a `row` with two `col` that contains 1 input text and the other 1 switch input.
```html
<div class="row">
   <div class="col-auto">
      <div class="mb-3">
         <label class="form-label">Text</label>
         <input type="text" class="form-control">     
      </div>
   </div>
   <div class="col">
      <div class="mb-3 form-switch">
         <div class="form-check">
            <input type="checkbox" class="form-check-input" value="1">
            <label class="form-check-label">Switch</label>
         </div>
      </div>
   </div>
</div>
```

![image](https://user-images.githubusercontent.com/25293190/204573358-f9357617-30c2-473d-9ccd-ff99c52fa1a1.png)

I tried to align them vertically but to each other.
By adding `align-items-end` class to the `row` and 30px height for the input `form-check`,
it works but feels kinda weird..
```diff
- <div class="row">
+ <div class="row align-items-end">
   <div class="col-auto">
      <div class="mb-3">
         <label class="form-label">Text</label>
         <input type="text" class="form-control">     
      </div>
   </div>
   <div class="col">
      <div class="mb-3 form-switch">
-         <div class="form-check">
+         <div class="form-check" style="height: 30px;">
            <input type="checkbox" class="form-check-input" value="1">
            <label class="form-check-label">Switch</label>
         </div>
      </div>
   </div>
</div>
```

![image](https://user-images.githubusercontent.com/25293190/204574212-1ce4ea47-5608-4781-a13e-106cc159739d.png)

Do you guys have a better way to align them ❓ 

`row align-items-center` doesn't work though:

![image](https://user-images.githubusercontent.com/25293190/204574032-195e08dc-c845-476c-a457-762682862c8b.png)

Thank you.

**Comment by JosephShown** [comment]
Ok so to explain this we have to discuss the responsive document flow and basically there are certain axis to snap to naturally as the align-items-end class is doing. The next stop in the axis is using align-items-center , which you noticed looks slightly off because of where it snaps to.  When making adjustments like this i recommend relational units like percentages vh vw for better responsiveness. 

My recommendation if i was trying to achieve what you are is to use padding or margin with percentages for the best result for all scenarios. Code is below.

```
 <div class="row align-items-end">
   <div class="col-auto">
      <div class="mb-3">
         <label class="form-label">Text</label>
         <input type="text" class="form-control">     
      </div>
   </div>
   <div class="col" style="padding-bottom: 1%;">
      <div class="mb-3 form-switch">
         <div class="form-check" style="">
            <input type="checkbox" class="form-check-input" value="1">
            <label class="form-check-label">Switch</label>
         </div>
      </div>
   </div>
</div>
```

**Reply by cavasinf** [reply]
I have no idea what you are trying to say 🤔 
`row` classes are already displayed as `flex`

**Comment by JosephShown** [comment]
I have updated my answer in order to better help you, please let me know if this works for you or you need additional help.
